### PR TITLE
morph: 1.6.0 -> 1.7.0

### PR DESCRIPTION
**Pull request by 0z13**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by arjan-s** [comment]
Result of `nixpkgs-review pr 169708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>morph</li>
  </ul>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 169708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>morph</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
morph:
<details>

got build log for '/nix/store/q831phfbvadx4y2gkli6axaca930y8pj-morph-1.7.0' from 'daemon'
When evaluating attribute ‘morph’:
warning: unused-argument
Unused argument: `go-bindata`.
Near pkgs/tools/package-management/morph/default.nix:1:35:

```
  |
1 | { buildGoModule, fetchFromGitHub, go-bindata, openssh, makeWrapper, lib }:
  |                                   ^
```
warning: unused-argument
Unused argument: `openssh`.
Near pkgs/tools/package-management/morph/default.nix:1:47:

```
  |
1 | { buildGoModule, fetchFromGitHub, go-bindata, openssh, makeWrapper, lib }:
  |                                               ^
```
warning: unused-argument
Unused argument: `makeWrapper`.
Near pkgs/tools/package-management/morph/default.nix:1:56:

```
  |
1 | { buildGoModule, fetchFromGitHub, go-bindata, openssh, makeWrapper, lib }:
  |                                                        ^
```
</details>
</details>

File: pkgs/tools/package-management/morph/default.nix [hunk 1]
```diff
@@ -2,33 +2,28 @@
...
+    ldflags+=" -X main.assetRoot=$lib"
   '';
 
   postInstall = ''
     mkdir -p $lib
     cp -v ./data/*.nix $lib
-    wrapProgram $out/bin/morph --prefix PATH : ${lib.makeBinPath [ openssh ]};
   '';
 
+  vendorSha256 = "08zzp0h4c4i5hk4whz06a3da7qjms6lr36596vxz0d8q0n7rspr9";
```
Comment (superherointj) [inline pkgs/tools/package-management/morph/default.nix:26]: Ideally `vendorSha256` should be bellow `src`.
And jump one line between phases.

File: pkgs/tools/package-management/morph/default.nix [hunk 2]
```diff
@@ -2,33 +2,28 @@
...
+    sha256 = "sha256-0CHmjqPxBgALGZYjfJFLoLBnoI0U7oZ8WyCtu1bkzZg=";
   };
 
-  vendorSha256 = "08zzp0h4c4i5hk4whz06a3da7qjms6lr36596vxz0d8q0n7rspr9";
-
-  nativeBuildInputs = [ makeWrapper go-bindata ];
-
   ldflags = [
     "-X main.version=${version}"
   ];
```
Comment (superherointj) [inline pkgs/tools/package-management/morph/default.nix:16]:
```suggestion
  ];

```

**Comment by gador** [comment]
Result of `nixpkgs-review pr 169708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>morph</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]

File: pkgs/tools/package-management/morph/default.nix [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
-    sha256 = "0aibs4gsb9pl21nd93bf963kdzf0661qn0liaw8v8ak2xbz7nbs8";
+    sha256 = "sha256-0CHmjqPxBgALGZYjfJFLoLBnoI0U7oZ8WyCtu1bkzZg=";
   };
 
   vendorSha256 = "08zzp0h4c4i5hk4whz06a3da7qjms6lr36596vxz0d8q0n7rspr9";
 
-  nativeBuildInputs = [ makeWrapper go-bindata ];
+  nativeBuildInputs = [ makeWrapper ];
 
   ldflags = [
```
Comment (superherointj) [inline pkgs/tools/package-management/morph/default.nix:18]: `ldflags` is configure phase. It comes before build phase.

Comment (superherointj) [reply pkgs/tools/package-management/morph/default.nix:18]: @tootoobeepbeep Please notice I'm not a `commiter`.
I help with reviews by checking trivial things  speed up the process.
But then a commiter has to step up to merge the PR.

File: pkgs/tools/package-management/morph/default.nix [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
     "-X main.version=${version}"
   ];
 
-  postPatch = ''
-    go-bindata -pkg assets -o assets/assets.go data/
+  preBuild = ''
+    ldflags+=" -X main.assetRoot=$lib"
   '';
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/morph/default.nix:24]: Please use tags.

Comment (SuperSandro2000) [inline pkgs/tools/package-management/morph/default.nix:24]:
```suggestion
```

File: pkgs/tools/package-management/morph/default.nix [hunk 5]
```diff
@@ -1,33 +1,33 @@
...
+    sha256 = "sha256-0CHmjqPxBgALGZYjfJFLoLBnoI0U7oZ8WyCtu1bkzZg=";
   };
 
   vendorSha256 = "08zzp0h4c4i5hk4whz06a3da7qjms6lr36596vxz0d8q0n7rspr9";
 
-  nativeBuildInputs = [ makeWrapper go-bindata ];
+  nativeBuildInputs = [ makeWrapper ];
 
   ldflags = [
     "-X main.version=${version}"
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/morph/default.nix:19]:
```suggestion
    "-X main.version=${version}"
    "-X main.assetRoot=${placeholder "lib"}"
```

File: pkgs/tools/package-management/morph/default.nix [hunk 6]
```diff
@@ -1,33 +1,30 @@
...
-  postPatch = ''
-    go-bindata -pkg assets -o assets/assets.go data/
-  '';
-
   postInstall = ''
     mkdir -p $lib
     cp -v ./data/*.nix $lib
     wrapProgram $out/bin/morph --prefix PATH : ${lib.makeBinPath [ openssh ]};
-  '';
+    '';
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/morph/default.nix:27]:
```suggestion
  '';
```

**Pull request merged**